server/issue.go

```diff
@@ -181,6 +178,126 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 	return http.StatusOK, nil
 }
 
+func httpAPIAttachCommentToIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodPost {
+		return http.StatusMethodNotAllowed,
+			errors.New("method " + r.Method + " is not allowed, must be POST")
+	}
+
+	api := ji.GetPlugin().API
+
+	comment := &struct {
+		PostId string       `json:"post_id"`
+		Fields jira.Comment `json:"fields"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&comment)
+	if err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request")
+	}
+
+	mattermostUserId := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	// Lets add a permalink to the post in the Jira Description
+	post, appErr := api.GetPost(comment.PostId)
+	if appErr != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(appErr, "failed to load post "+comment.PostId)
+	}
+	if post == nil {
+		return http.StatusInternalServerError,
+			errors.New("failed to load post " + comment.PostId + ": not found")
+	}
+
+	commentUser, appErr := api.GetUser(post.UserId)
+	if appErr != nil {
+		return http.StatusInternalServerError,
+			errors.New("failed to load post.UserID " + post.UserId + ": not found")
+	}
+
+	permalink, err := getPermaLink(ji, comment.PostId, post)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.New("failed to get permalink for " + comment.PostId + ": not found")
+	}
+
+	permalinkMessage := fmt.Sprintf("*@%s attached a* [message|%s] *from @%s*\n", jiraUser.User.Name, permalink, commentUser.Username)
+
+	if len(comment.Fields.Body) > 0 {
+		comment.Fields.Body += fmt.Sprintf("\n%v", permalinkMessage)
+	} else {
+		comment.Fields.Body = permalinkMessage
+	}
+	comment.Fields.Body += post.Message
+
+	commentAdded, _, err := jiraClient.Issue.AddComment(comment.Fields.ID, &comment.Fields)
```

Comment:
The first argument to `AddComment` is the issue Id. It's a bit confusing to see it stored in `Fields.ID`

Reply:
Agree and am making changes to the fields of the struct and variable naming so make this clearer.